plugins/exporternew.koplugin/formats/base.lua

```diff
@@ -0,0 +1,70 @@
+local DataStorage = require("datastorage")
+
+local BaseExporter = {}
+
+function BaseExporter:new(o)
+    o = o or {}
+    o.id = "exporter_new"
+    o.name = o.name or "generic"
+    o.extension = o.extension or o.name or "export"
+    o.clipping_dir = DataStorage:getDataDir() .. "/clipboard"
+    o.is_remote = o.is_remote or false
+    setmetatable(o, self)
+    self.__index = self
+    return o:init()
+end
+
+function BaseExporter:init()
+    self:loadSettings()
+
+    return self
+end
+
+function BaseExporter:isEnabled()
+    return self.settings.enabled
+end
+
+function BaseExporter:toggleEnabled()
+    self.settings.enabled = not self.settings.enabled
+    self:saveSettings()
+end
+
+function BaseExporter:loadSettings()
+    local plugin_settings = G_reader_settings:readSetting(self.id) or {}
+    self.settings = plugin_settings[self.name] or {}
+end
+
+function BaseExporter:saveSettings()
+    local plugin_settings = G_reader_settings:readSetting(self.id) or {}
+    plugin_settings[self.name] = self.settings
+    G_reader_settings:saveSetting(self.id, plugin_settings)
+end
+
+function BaseExporter:exportOne(t) end
+function BaseExporter:exportAll(t) end
+
+function BaseExporter:getFileTimeStamp(timestamp)
```

Comment:
```suggestion
function BaseExporter:getTimeStamp()
    local ts = self.timestamp or os.time()
    return os.date("%Y-%m-%d %H:%M:%S", ts)
end
```

Reply:
I was suppying timestamp from the upper level so that if one chooses to export in multiple format then they get same timestamp.

We can do things this way too. But if we do so, I would like to create two fuctions. `getTimestamp` and `formatTimestampForFile`. One can choose different representations in different places.

Reply:
@pazos
Should we implement the behavior you're implying or will keep the one I've implemented?

Reply:
I've implemented this with the modifications suggested in this thread. I did that because if I do, in the JSON export I can use that timestamp as a UNIX timestamp for `created_on` in all book export. String format would've been inconsistent with the rest of the file.